autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_dataset.py

```diff
@@ -114,29 +116,34 @@ def __init__(self, processed_array, feature_arraycol_map, feature_type_map, batc
             for feature in feature_type_map:
                 if feature_type_map[feature] == 'embed':
                     feature_colind = feature_arraycol_map[feature]
-                    data_list.append(mx.nd.array(processed_array[:,feature_colind], dtype='int32')) # array of ints with data for this embedding feature 
+                    data_list.append(mx.nd.array(processed_array[:,feature_colind], dtype='int32')) # array of ints with data for this embedding feature
                     self.data_desc.append("embed")
                     self.feature_dataindex_map[feature]  = len(data_list)-1
 
         if len(self.feature_groups['language']) > 0:
             for feature in feature_type_map:
                 if feature_type_map[feature] == 'language':
                     feature_colinds = feature_arraycol_map[feature]
-                    data_list.append(mx.nd.array(processed_array[:,feature_colinds], dtype='int32')) # array of ints with data for this language feature 
+                    data_list.append(mx.nd.array(processed_array[:,feature_colinds], dtype='int32')) # array of ints with data for this language feature
                     self.data_desc.append("language")
                     self.feature_dataindex_map[feature]  = len(data_list)-1
 
         self.num_classes = None
         if labels is not None:
             labels = np.array(labels)
-            if problem_type == REGRESSION and labels.dtype != np.float32:
-                    labels = labels.astype('float32') # Convert to proper float-type if not already
-            data_list.append(mx.nd.array(labels.reshape(len(labels),1)))
             self.data_desc.append("label")
-            self.label_index = len(data_list) - 1 # To access data labels, use: self.dataset._data[self.label_index]
-            if problem_type in [BINARY, MULTICLASS]:
-                self.num_classes = len(set(labels))
-        
+            self.label_index = len(data_list) # To access data labels, use: self.dataset._data[self.label_index]
+            self.num_classes = None
+            if self.problem_type == REGRESSION and labels.dtype != np.float32:
```

Comment:
do you mean the `self.num_classes = None` or the `if` statement.  If you mean the latter, wouldn't the readability be most improved if I simply change L139 to: 

`elif self.problem_type == SOFTCLASS:`

Reply:
if self.problem_type == REGRESSION and labels.dtype != np.float32: -> Line 143, since it will only be able to be triggered if the else statement would have triggered. I mean the if statement.